jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -224,18 +254,29 @@ public XmlConfiguration(String configuration) throws SAXException, IOException
      * @param configuration An input stream containing a complete configuration file
      * @throws IOException if the configuration could not be read
      * @throws SAXException if the configuration could not be parsed
+     * @deprecated use Constructor which has location information
      */
+    @Deprecated
     public XmlConfiguration(InputStream configuration) throws SAXException, IOException
     {
         InputSource source = new InputSource(configuration);
         synchronized (__parser)
         {
-            _url = null;
+            _location = null;
             setConfig(__parser.parse(source));
             _dtd = __parser.getDTD();
         }
     }
 
+    private String getLocation()
+    {
+        if (_location == null)
+        {
+            return "UNKNOWN-LOCATION";
+        }
+        return _location.toString();
+    }
+
```

Comment:
I don't like this method because it seems a legit getter for property `location`, but it cannot be used as such (all places that need the "location" for processing purposes needs to reference the field directly) - in fact it is used only for printing purposes.
I suggest that this method is renamed to `XmlConfiguration.toLocation()` or something similar to (more or less) indicate that it is not a getter and its purpose is for printing (akin to `Object.toString()`).

Reply:
Fair enough.  I'll make that change.

Reply:
Done.